src/main/java/org/spongepowered/mod/mixin/world/MixinWorld.java

```diff
@@ -260,4 +328,94 @@ public boolean doesKeepSpawnLoaded() {
+        this.playSound(sound, position, volume, pitch, 0);
+    }
+
+    @Override
+    public void playSound(SoundType sound, Vector3d position, double volume, double pitch, double minVolume) {
+        this.playSoundEffect(position.getX(), position.getY(), position.getZ(), sound.getName(), (float) Math.max(minVolume, volume), (float) pitch);
+    }
+
+    @Override
+    public Optional<Entity> getEntityFromUUID(UUID uuid) {
+        if ((Object) this instanceof WorldServer) {
+            // TODO Should this be done in an override in a WorldServer mixin?
+            return Optional.fromNullable((Entity) ((WorldServer) (Object) this).getEntityFromUuid(uuid));
+        }
+        for (net.minecraft.entity.Entity entity : this.loadedEntityList) {
+            if (entity.getUniqueID().equals(uuid)) {
+                return Optional.of((Entity) entity);
+            }
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<String> getGameRule(String gameRule) {
+        if (this.worldInfo.getGameRulesInstance().hasRule(gameRule)) {
+            return Optional.of(this.worldInfo.getGameRulesInstance().getGameRuleStringValue(gameRule));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public void setGameRule(String gameRule, String value) {
+        this.worldInfo.getGameRulesInstance().setOrCreateGameRule(gameRule, value);
+    }
+
+    @Override
+    public Map<String, String> getGameRules() {
+        GameRules gameRules = this.worldInfo.getGameRulesInstance();
+        // TODO Could expose GameRules#theGameRules via reflection to remove the
+        // need to iterate over keys
+        Map<String, String> ruleMap = new HashMap<String, String>();
+        for (String rule : gameRules.getRules()) {
+            ruleMap.put(rule, gameRules.getGameRuleStringValue(rule));
+        }
+        return ruleMap;
+    }
+
+    @Override
+    public long getWorldSeed() {
+        return this.getSeed();
+    }
+
+    @Override
+    public void setSeed(long seed) {
+        ObfuscationReflectionHelper.setPrivateValue(WorldInfo.class, this.worldInfo, seed, "randomSeed");
+        this.rand.setSeed(seed);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Iterable<Chunk> getLoadedChunks() {
+        // Should this assume provider is ChunkProviderServer or would this
+        // method be implemented in a mixin with WorldServer?
+        IChunkProvider provider = this.getChunkProvider();
+        if (provider instanceof ChunkProviderServer) {
+            return ((ChunkProviderServer) provider).loadedChunks;
+        }
+        return Collections.EMPTY_LIST;
+    }
+
+    @Override
+    public boolean deleteChunk(Chunk chunk) {
+        IChunkProvider provider = this.getChunkProvider();
+        Vector3i chunkPos = chunk.getPosition();
+        if (provider instanceof ChunkProviderServer) {
```

Comment:
Every worldserver creates a ChunkProviderServer instance, there is no need to check anything else. There also needs to be a check for forced chunks as many forge mods contain blocks that keep chunks loaded. If a forced chunk is found, this method needs to return false.